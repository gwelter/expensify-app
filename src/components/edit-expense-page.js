import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './expense-form.js';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
    return (
        <div>
            Editing expense with id {props.match.params.id}
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense) => {
                    console.log(expense);
                    props.dispatch(editExpense(props.match.params.id, expense));
                    props.history.push('/');
                }}
            />
            <button onClick={() => {
                props.dispatch(removeExpense({id: props.match.params.id}));
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
};

const mapStateToProps = (state, props) => ({expense: state.expenses.find((expense) => expense.id === props.match.params.id)});

export default connect(mapStateToProps)(EditExpensePage);