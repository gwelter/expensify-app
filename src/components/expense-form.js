import React from 'react';

export default class ExpenseForm extends React.Component {
	state = {
		description: '',

	};
	onDescriptionChange = (e) => {
		const description = e.target.value;
		this.setState(() => ({ description }));
	};
	render() {
		return (
			<div>
				<form>
					<input
						type="text"
						placeholder="Description"
						value={this.state.description}
						onChange={this.onDescriptionChange}
						autoFocus />
					<input
						type="number"
						placeholder="Amount" />
					<textarea placeholder="Add a note for your expense (optional)">
					</textarea>
					<button>Add Expense</button>
				</form>
			</div>
		);
	}
};