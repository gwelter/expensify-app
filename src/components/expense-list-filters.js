import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };
    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={(e) => {
                    this.props.dispatch(setTextFilter(e.target.value));
                }}/>
                <select
                value={this.props.filters.sortBy}
                onChange={(e) => {
                    if(e.target.value === "date") {
                        this.props.dispatch(sortByDate());
                    } else if(e.target.value === "amount") {
                        this.props.dispatch(sortByAmount());
                    }
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    startDateId="your_unique_start_date_id"
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    endDateId="your_unique_end_date_id"
                    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={calendarFocused => this.setState({ calendarFocused })} // PropTypes.func.isRequired,
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

export default connect(mapStateToProps)(ExpenseListFilters);