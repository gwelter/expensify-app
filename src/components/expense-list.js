import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './expense-list-item';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense) => (
            <ExpenseListItem key={expense.id} {...expense} />
        ))}
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);

