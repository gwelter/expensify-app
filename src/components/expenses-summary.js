import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
    <div>
        {expensesCount !== 0 && <h2>Viewing {expensesCount} expenses totalling {numeral(expensesTotal / 100).format('$0,00.00')}</h2>}
    </div>
);

const mapStateToProps = (state) => {
    let filteredExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: filteredExpenses.length,
        expensesTotal: selectExpensesTotal(filteredExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);