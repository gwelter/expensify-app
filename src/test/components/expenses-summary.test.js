import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/expenses-summary';

test('should correctly render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={240}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={1231241241}/>);
    expect(wrapper).toMatchSnapshot();
});