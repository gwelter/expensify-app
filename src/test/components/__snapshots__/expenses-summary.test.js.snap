// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    23
     expenses totalling 
    $12,312,412.41
  </h2>
</div>
`;

exports[`should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     expenses totalling 
    $2.40
  </h2>
</div>
`;
